public/app/core/services/bridge_srv.ts

```diff
@@ -62,6 +65,18 @@ export class BridgeSrv {
         });
         console.log('store updating angular $location.url', url);
       }
+
+      // Check for template variable changes
+      if (this.lastQuery) {
+        const changes = findTemplateVarChanges(state.location.query, this.lastQuery);
+        if (changes) {
+          const dash = getDashboardSrv().getCurrent();
+          if (dash) {
+            dash.events.emit(CoreEvents.templateVarsChangedInUrl, changes);
```

Comment:
technically bridgeSrv could just inject variableSrv and call templateVarsChangedInUrl as both are angular services, but this is fine as well